lib/fastlane_core/configuration/config_item.rb

```diff
@@ -63,16 +66,37 @@ def valid?(value)
 
     # Returns an updated value type (if necessary)
     def auto_convert_value(value)
-      # Special treatment if the user specififed true, false or YES, NO
-      if %w(YES yes true).include?(value)
-        value = true
-      elsif %w(NO no false).include?(value)
-        value = false
+      # We must cast the type to String before comparing because of the way that Ruby
+      # handles class comparisons
```

Comment:
I wanted to stick with a `case` statement since it would be much more extensible if more logic is added in the future to support conversion for other data types.

The original code was:

```ruby
case data_type
when Array
  # do stuff
else
  # do other stuff
end
```

Because of the way Ruby does comparisons using the `===` method (which `case` statements use), that doesn't work. However, casting the data type (which is a `Class`) to string allows for comparison.

Another option is:

```ruby
case
when value.kind_of?(Array)
  # do stuff
else
  # do other stuff
end
```

That's pretty much just rewriting a chained if statement, so I don't see much value in that approach. At least the first approach is a bit more extensible. Thoughts?